rust-runtime/aws-smithy-http/src/middleware.rs

```diff
@@ -111,12 +129,14 @@ where
     };
 
     let http_response = http::Response::from_parts(parts, Bytes::from(body));
-    trace!(http_response = ?http_response);
-    let parsed = handler.parse_loaded(&http_response);
-    sdk_result(
-        parsed,
-        operation::Response::from_parts(http_response.map(SdkBody::from), properties),
-    )
+    trace!(http_response = ?http_response, "read HTTP response body");
```

Comment:
I'm thinking we should we `debug!` the request ID header if one is present. Or does hyper already log the headers?

Reply:
`?http_response` will output all the headers, but it's at `trace` level. I think request IDs are an AWS concept though? Or at least, this header would be specific to AWS? So this is probably not the right place to add it.

Reply:
I would support adding a default middleware to log this stuff though...or maybe expose it in the property bag?